pixels/utils/auth.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import secrets
+
+from asyncpg import Connection
+from fastapi import HTTPException, Request
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from jose import JWTError, jwt
+
+from pixels.constants import Server
+from pixels.models import AuthState
+
+log = logging.getLogger(__name__)
+
+
+class JWTBearer(HTTPBearer):
+    """Dependency for routes to enforce JWT auth."""
+
+    def __init__(self, auto_error: bool = True, is_mod_endpoint: bool = False):
+        super().__init__(auto_error=auto_error)
+        self.is_mod_endpoint = is_mod_endpoint
+
+    async def __call__(self, request: Request):
+        """Check if the supplied credentials are valid for this endpoint."""
+        credentials: HTTPAuthorizationCredentials = await super().__call__(request)
+        credentials = credentials.credentials
+        if not credentials:
+            raise HTTPException(status_code=403, detail=AuthState.NO_TOKEN)
+
+        try:
+            token_data = jwt.decode(credentials, Server.JWT_SECRET)
+        except JWTError:
+            raise HTTPException(status_code=403, detail=AuthState.INVALID_TOKEN.value)
+
+        user_id = token_data["id"]
+        token_salt = token_data["salt"]
```

Comment:
Just syntactic sugar
```suggestion
        user_id, token_salt = token_data["id"], token_data["salt"]
```